core/src/main/java/org/apache/gravitino/storage/relational/service/TableMetaService.java

```diff
@@ -229,6 +231,31 @@ public boolean deleteTable(NameIdentifier identifier) {
           if (deleteResult.get() > 0) {
             TableColumnMetaService.getInstance().deleteColumnsByTableId(tableId);
           }
+        },
+        () -> {
+          if (deleteResult.get() > 0) {
```

Comment:
This is a problem unrelated to this PR. please see L218, Since it uses `doWithCommitAndFetchResult`, so it's will use two transactions in the `deleteTable` if I'm not wrong.

Reply:
Yes, we should change this.